### PR TITLE
Better UEFI extractor

### PR DESCRIPTION
**Pull request by BenediktMKuehne**
**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Add a more general extraction tool for UEFI images
+ meta-data extraction from UEFI

**What is the current behavior?** (You can also link to an open issue here)
1. either EFI images get extracted via last-resort (binwalk)
2. or AMI-extractor works

**What is the new behavior (if this is a feature change)? If possible add a screenshot.**
UEFI images get extracted with [UEFITool](https://github.com/LongSoft/UEFITool)

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
❗  **Requires new EMBA container**

### CODE REVIEW

File: installer/IP35_uefi_extraction.sh [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+#
+# EMBA comes with ABSOLUTELY NO WARRANTY. This is free software, and you are
+# welcome to redistribute it under the terms of the GNU General Public License.
+# See LICENSE file for usage of this software.
+#
+# EMBA is licensed under GPLv3
+#
+# Author(s): Benedikt Kuehne
+
+# Description:  Installs UEFIExtract from https://github.com/LongSoft/UEFITool/releases
```
Comment (BenediktMKuehne) [inline installer/IP35_uefi_extraction.sh:15]: Q: Static release version ok or should I switch to build from source?

Comment (m-1-k-3) [reply installer/IP35_uefi_extraction.sh:15]: As long as it works on Kali this is fine

File: modules/P35_UEFI_extractor.sh [hunk 2]
```diff
@@ -75,3 +77,62 @@ ami_extractor() {
...
+  print_ln
+
+  NVARS=$(grep -c "NVAR entry" "$UEFI_EXTRACT_REPORT_FILE")
+  PE32_IMAGE=$(grep -c "PE32 image" "$UEFI_EXTRACT_REPORT_FILE")
+  EFI_ARCH=$(find "$EXTRACTION_DIR_" -name 'info.txt' -exec grep 'Machine type:' {} \; | sed -E 's/Machine\ type\:\ //g' | uniq )
+
+  if [[ -n "$EFI_ARCH" ]]; then
+    print_output "[*] Found $ORANGE$PE32_IMAGE$NC PE32 images for architecture $ORANGE$EFI_ARCH$ORANGE drivers."
+    print_output "[+] Possible architecture details found ($ORANGE UEFI Extractor $NC): $ORANGE$EFI_ARCH$NC"
+    export EFI_ARCH
```
Comment (BenediktMKuehne) [inline modules/P35_UEFI_extractor.sh:127]: Since this is basically the same as `file`-check, should we set this to verified $ARCH ?

Comment (m-1-k-3) [reply modules/P35_UEFI_extractor.sh:127]: Please do some tests if we can trust this architecture detection

Comment (BenediktMKuehne) [reply modules/P35_UEFI_extractor.sh:127]: will do

File: helpers/helpers_emba_dependency_check.sh [hunk 3]
```diff
@@ -383,7 +383,10 @@ dependency_check()
 
     check_dep_tool "ubireader image extractor" "ubireader_extract_images"
     check_dep_tool "ubireader file extractor" "ubireader_extract_files"
-
+    
+    # UEFI
+    check_dep_tool "UEFI image extractor" "$EXT_DIR""/UEFITool/UEFIExtract"
```
Comment (m-1-k-3) [inline helpers/helpers_emba_dependency_check.sh:388]: could you comment this until we have the pre-requ in our container?

**Comment by m-1-k-3** [comment]
Are we ready for review and testing?

**Review by m-1-k-3 — APPROVED** [review]

**Comment by m-1-k-3** [comment]
I will do some further tests and rebuild the container before merge.

File: installer/IP35_uefi_extraction.sh [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      if ! [[ -d external/UEFITool ]]; then
+          mkdir external/UEFITool
+      fi
+      download_file "UEFIExtract_NE_A62_linux_x86_64.zip" "https://github.com/LongSoft/UEFITool/releases/download/A62/UEFIExtract_NE_A62_linux_x86_64.zip" "external/UEFITool/UEFIExtract_NE_A62_linux_x86_64.zip"
+      if [[ -f "external/UEFITool/UEFIExtract_NE_A62_linux_x86_64.zip" ]]; then
+        if ! [[ -f external/UEFITool/UEFIExtract ]]; then
+          unzip external/UEFITool/UEFIExtract_NE_A62_linux_x86_64.zip -d external/UEFITool
+        fi
+      else
+          echo -e "$ORANGE""UEFITool installation failed - check it manually""$NC"
```
Comment (m-1-k-3) [inline installer/IP35_uefi_extraction.sh:46]: so many spaces

**Comment by m-1-k-3** [comment]
Sorry but I missed the installation in the last container build. Will start now again with building :)

File: installer/IP35_uefi_extraction.sh [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+IP35_uefi_extraction() {
+  module_title "${FUNCNAME[0]}"
+  
+  
+  INSTALL_APP_LIST=()
+
+  print_file_info "UEFIExtract_NE_A62_linux_x86_64.zip" "Release-version A62" "https://github.com/LongSoft/UEFITool/releases/download/A62/UEFIExtract_NE_A62_linux_x86_64.zip" "external/UEFITool/UEFIExtract_NE_A62_linux_x86_64.zip"
+  print_tool_info "unzip" 1
+
+  if [[ "$LIST_DEP" -eq 1 ]] || [[ $DOCKER_SETUP -eq 0 ]]; then
```
Comment (m-1-k-3) [inline installer/IP35_uefi_extraction.sh:27]: DOCKER_SETUP is for building the container. Your check bypasses this and the new dependency will not get installed in the container.
I guess this should be something like this:
```
    if [[ "$LIST_DEP" -eq 1 ]] || [[ $DOCKER_SETUP -eq 1 ]] ; then
```

**Review by m-1-k-3 — APPROVED** [review]

**Pull request merged**